compiler/vm/vmmarshal.nim

```diff
@@ -0,0 +1,242 @@
+## This module contains functions for converting between the PNode
+## based data representation and the flat memory one the VM uses
+
+# TODO: "marshaling" doesn't feel like the correct description

```

Comment:
How about vmcompilerffi or vmcompilerdata or vmcompilerserdes?

serdes: serialization and deserialization

Reply:
`vmcompilerserdes` it is :)